VAULT-14644 Add support for Azure WIF auth to auto-auth (for Agent and Proxy) (#22264)
hashicorp/vault: website/content/docs/agent-and-proxy/autoauth/methods/azure.mdx

- `role` `(string: required)` - The role to authenticate against on Vault

- `authenticate_from_environment` `(boolean: false)` - If set to true, will attempt to make an Azure
  identity object through Azure specific environment variables, in priority order and as specified [in
  Azure documentation][environment-variables]. Will work seamlessly with Azure
  [Workload Identity][workload-identity] and others. If set to true, `object_id`, and

>   `client_id` will be ignored in favour of this authentication approach.

Reviewer: ❓ Do you think we should add a `~> Note:`  for this part:

> If set to true, `object_id`, and `client_id` will be ignored in favour of this authentication approach.

Author: Good idea! I didn't think that was common for config value lists like this, but it seems it's done in other places

> - `scope` `(string: optional)` - The scope to use when obtaining the token. If not given, will be
>   the value used for `resource` appended with `/.default` appended to it. For more information on
>   scopes in Azure, see [this document][scopes].

Reviewer: ```suggestion
- `scope` `(string: optional)` - The scope to use when obtaining a token. If not supplied, will default to 
  the value used for `resource` with "`/.default`" appended as a suffix. For more information on
  scopes in Azure, see [this document][scopes].
```

Author: Thanks! Will add this. Much better wording.

  when acquiring an [access token][azure-access-token]. Only one of `object_id` or `client_id`
  may be provided.

## Example configuration

> An example auto-auth configuration block for an example Azure [Workload Identity][workload-identity] configured client follows:

Reviewer: ```suggestion
Example client 'auto-auth' configuration block for an Azure [Workload Identity][workload-identity]:
```

Author: Looks great! Thanks :)
